lib/oracle.js

```diff
@@ -150,61 +179,93 @@ Oracle.prototype.executeSQL = function (sql, params, options, callback) {
+        }
+        if (self.settings.debug && data) {
+          self.debug("Result: %j", data);
+        }
+        if (data && data.rows) {
+          data = data.rows;
+        }
+        callback(err, data);
+      });
     return;
   }
 
-  /*
-  self.pool.getConnection(function (err, connection) {
+  self.pool.getConnection(function(err, connection) {
     if (err) {
       callback && callback(err);
       return;
     }
-    console.log('open %d -> %d', open, ++open);
     if (self.settings.debug) {
-      debugConnection('Connection acquired: ', self.pool.getInfo());
+      self.debug('Connection acquired: ', self.pool);
     }
-    connection.execute(sql, params, function (err, data) {
-      if (err && self.settings.debug) {
-        self.debug(err);
-      }
-      if (self.settings.debug && data) {
-        self.debug("Result: %j", data);
-      }
-      if (log) {
-        log(sql, time);
-      }
-      connection.close(function(err) {
-        if (err) {
-          return callback(err);
+    connection.clientId = self.settings.clientId || 'LoopBack';
+    connection.module = self.settings.module || 'loopback-connector-oracle';
+    connection.action = self.settings.action || '';
+
+    connection.execute(sql, params,
+      {outFormat: oracle.OBJECT, autoCommit: true},
+      function(err, data) {
+        if (err && self.settings.debug) {
+          self.debug(err);
         }
-        console.log('close %d -> %d', open, --open);
-        if (self.settings.debug) {
-          debugConnection('Connection released: ', self.pool.getInfo());
+
+        if (!err && data) {
+          var lobs = [];
+          if (data.rows) {
+            data = data.rows;
+            if (self.settings.debug && data) {
+              self.debug("Result: %j", data);
+            }
+            if (Array.isArray(data)) {
+              for (var i = 0, n = data.length; i < n; i++) {
+                // Iterate through the rows
+                var row = data[i];
+                for (var k in row) {
+                  var val = row[k];
+                  if (val instanceof stream.Readable) {
+                    lobs.push({index: i, key: k, stream: val});
+                  }
+                }
+              }
+            }
+          }
+          if (lobs.length === 0) {
+            releaseConnectionAndCallback();
+            return;
+          }
+          async.each(lobs, function(lob, done) {
+            readStream(lob.stream, function(err, d) {
+              if (err) return done(err);
+              data[lob.index][lob.key] = d;
+              done();
+            });
+          }, function(err) {
+            if (err) return callback(err);
+            releaseConnectionAndCallback();
+          });
+        } else {
+          releaseConnectionAndCallback();
+        }
+
+        function releaseConnectionAndCallback() {
+          connection.release(function(err) {
+            if (err) {
+              self.debug(err);
```

Comment:
Ditto.
Maybe add a name for debug message like `self.debug('Release error: ' + err)`.